extensions/wikia/PhalanxII/models/PhalanxModel.php

```diff
@@ -33,7 +31,7 @@ public function __construct() {
 		parent::__construct();
 
 		$this->user = $this->wg->user;
-		$this->service = Injector::getInjector()->get( PhalanxService::class );
+		$this->service = PhalanxServiceFactory::getServiceInstance();
 		$this->ip = $this->wg->request->getIp();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app